app/controllers/taxpayer/WhatIsTaxpayersStartDateForImplementingArrangementController.scala

```diff
@@ -0,0 +1,121 @@
+    formProvider: WhatIsTaxpayersStartDateForImplementingArrangementFormProvider,
+    val controllerComponents: MessagesControllerComponents,
+    renderer: Renderer
+)(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with NunjucksSupport {
+
+  val numberOfMonthsToAdd = 6
+  val form = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      val preparedForm = request.userAnswers.get(WhatIsTaxpayersStartDateForImplementingArrangementPage) match {
+        case Some(value) => form.fill(value)
+        case None        => form
+      }
+
+      val viewModel = DateInput.localDate(preparedForm("value"))
+
+      getDisplayName(request.userAnswers) match {
+        case Some(displayName) =>
+          val json = Json.obj (
+          "form" -> preparedForm,
+          "mode" -> mode,
+          "date" -> viewModel,
+          "exampleDate" -> LocalDate.now.plusMonths (numberOfMonthsToAdd).format (dateFormatterNumericDMY),
+          "displayName" -> getDisplayName (request.userAnswers)
+          )
+
+          renderer.render ("taxpayer/whatIsTaxpayersStartDateForImplementingArrangement.njk", json).map (Ok (_) )
+        case None => Future.successful(Redirect(controllers.taxpayer.routes.TaxpayerSelectTypeController.onPageLoad(mode)))
+      }
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      form.bindFromRequest().fold(
+        formWithErrors =>  {
+
+          val viewModel = DateInput.localDate(formWithErrors("value"))
+
+          val json = Json.obj(
+            "form" -> formWithErrors,
+            "mode" -> mode,
+            "date" -> viewModel,
+            "exampleDate" -> LocalDate.now.plusMonths(numberOfMonthsToAdd).format(dateFormatterNumericDMY),
+            "displayName" -> getDisplayName(request.userAnswers)
+          )
+
+          renderer.render("taxpayer/whatIsTaxpayersStartDateForImplementingArrangement.njk", json).map(BadRequest(_))
+        },
+        value =>
+          for {
+            updatedAnswers <- Future.fromTry(request.userAnswers.set(WhatIsTaxpayersStartDateForImplementingArrangementPage, value))
+            _              <- sessionRepository.set(updatedAnswers)
+          } yield Redirect(navigator.nextPage(WhatIsTaxpayersStartDateForImplementingArrangementPage, mode, updatedAnswers))
+      )
+  }
+
+  private def getDisplayName(userAnswers: UserAnswers): Option[String] = {
+   userAnswers.get(TaxpayerSelectTypePage) map {
+     case Organisation => userAnswers.get(OrganisationNamePage) match {
+       case Some(organisationName) => organisationName
+       case None => "the organisation"
+     }
+     case Individual => userAnswers.get(IndividualNamePage) match {
+       case Some(individualName) => individualName.displayName
+       case None => "the individual"
```

Comment:
You can also call the getIndividualName and getOrganisationName in JourneyHelpers